docs/design/proposals/rbd-mirror.md

```diff
@@ -0,0 +1,137 @@
+# RBD MIRRORING
+
+![mirror](https://i.imgur.com/YF5WeCV.png)
+
+RBD images can be asynchronously mirrored between two Ceph clusters.
+
+## Challenges in ceph-csi
+
+* Currently, ceph-csi generates its unique ID for each rbd image and stores the
+  corresponding PV name and the unique ID mapping, and creates the rbd image
+  with the unique ID and return the encoded value which contains all the
+  required information for other operations. As the same rbd image needs to be
+  used in remote OCP cluster and the remote ceph cluster. In ceph-csi, we need
+  to change the logic of storing/generating the unique ID based on the PVC name
+  and namespace name.
+
+* ceph-csi generated journal(omap data) is not mirrored to the remote cluster.
+
+* Promotion and demotion of RBD image.
+
+## Below provides the outline of how this can be implemented in ceph-csi
+
+* A storage-class need to have a parameter to tell that the image needs to be
+  mirrored.
+* The PVC name and namespace need to be passed to the CSI driver as part of
+  volume create request (rbd image and omap keys/value will be derived from
+  these names if its mirrored image.
+* The RBD mirror daemon required setup has to be done prior enabling the image
+  mirroring from ceph-csi.
+* Promotion and Demotion of RBD image.
+
+### Storage-class parameter
+
+we need to add a new parameter to the storage-class to indicate the image need
```

Comment:
It'd be nice to explain that we can have multiple SC serving the same pool but bringing different mirroring capabilities.